src/main/single-habit/single-habit.component.html

```diff
@@ -0,0 +1,63 @@
+<div *ngIf="!isResize">
+  <div #habit (click)="updateProgressView(habit,$event)" (contextmenu)="openEditHabitMenu($event)"
+       *ngFor="let habit of getFilteredHabits()" class="habit">
+    <div class="habit-identification">
+      <img [src]="getHabitIcon(habit)" alt="habit" class="icon">
+    </div>
+    <div class="habit-content">
+      <div class="minor-details">
+        <div [ngClass]="{'canceled-text': habit.isFailed || habit.isSkipped || habit.isCompleted}" class="habit-name">
+          {{habit.name}}
+        </div>
+        <div [ngClass]="{'canceled-text': habit.isFailed || habit.isSkipped || habit.isCompleted}"
+             class="habit-frequency">
+          {{getSingleNumber(habit)}} / 1 times
+        </div>
+      </div>
+      <button (click)="completeHabit(habit)" *ngIf="showDoneButton(habit)" class="done-button" pButton type="button">
+        <i class="pi pi-check" style="font-size: 0.88rem"></i>
+        <span class="done">Done</span>
+      </button>
+      <button (click)="toggleLogValueBar(habit)" *ngIf="showLogButton(habit)" class="log-button" pButton type="button">
+        <img alt="log" class="log-icon"
+             src="assets/svg/log.svg"/>
+        <span class="log">Log</span>
+      </button>
+
+      <button *ngIf="habit.showLogValueBar" class="log-habit-bar" pButton type="button">
+        <input autofocus class="log-input" placeholder="Value">
+        <span class="log-text">km</span>
+        <img alt="log" class="log-tick-icon"
+             src="assets/svg/small-tick.svg"/>
+        <img (click)="closeLogValueBar(habit)" alt="log"
+             class="log-cross-icon" src="assets/svg/cross.svg"/>
+      </button>
+
+      <button #more
+              *ngIf="!habit.isCompleted && !habit.isSkipped && !habit.isFailed"
+              [matMenuTriggerFor]="editHabit.getMenu()" class="more-button" pButton type="button">
+        <i class="pi pi-ellipsis-v more-icon" style="font-size: 0.9rem"></i>
+      </button>
+      <button *ngIf="habit.isCompleted" [matMenuTriggerFor]="compHabit.getMenu()" class="more-button" pButton
+              type="button">
+        <i class="pi pi-ellipsis-v more-icon" style="font-size: 0.9rem"></i>
+      </button>
+      <button *ngIf="habit.isSkipped" [matMenuTriggerFor]="skippedHabit.getMenu()" class="more-button" pButton
+              type="button">
+        <i class="pi pi-ellipsis-v more-icon" style="font-size: 0.9rem"></i>
+      </button>
+      <button *ngIf="habit.isFailed" [matMenuTriggerFor]="failedHabit.getMenu()" class="more-button" pButton
+              type="button">
+        <i class="pi pi-ellipsis-v more-icon" style="font-size: 0.9rem"></i>
```

Comment:
we can create a class for this style and then use it for all other places as well